src/controllers/RoleAndPermission.controller.ts

```diff
@@ -0,0 +1,204 @@
+import { Request, Response } from "express";
+import { UserAttributes } from "../interfaces";
+import { RoleAndPermissionAttribute } from "../interfaces/Roles&Permission.interface";
+import { findOneUserService } from "../services";
+import {
+  deleteRoleService,
+  getAllRolesAndPermissons,
+  getRoleAndPermisson,
+  setRoleAndPermissionServise,
+  setUserRoleService,
+  updatePermissionsService,
+} from "../services/RolesAndPermission.service";
+
+// Admin set role to a particular user
+export const setRoleToUser = async (req: Request, res: Response) => {
+  try {
+    const wantedUser = req.params.id;
+    const oneUser = await findOneUserService(wantedUser);
+
+    if (oneUser) {
+      const parsedOneUser: UserAttributes = JSON.parse(JSON.stringify(oneUser));
+
+      parsedOneUser.role = req.body.role;
+
+      const updatedUserRole = await setUserRoleService(
+        parsedOneUser,
+        wantedUser,
+      );
+
+      return res.status(200).json({
+        status: 200,
+        success: true,
+        data: updatedUserRole,
+      });
+    } else {
+      res.status(404).json({
+        status: 404,
+        success: false,
+        message: "User not found",
+      });
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while seting role and permission",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// Admin set permissions w.r.t role
+export const setRoleAndPermission = async (req: Request, res: Response) => {
+  try {
+    const roleAndPermissonInfo = req.body as RoleAndPermissionAttribute;
+
+    if (await getRoleAndPermisson(req.body.role)) {
```

Comment:
May you first sanitize this. 🗑️ 

As it is, you don't know the shape of this input. e.g whether it's a boolean, number, string...

Reply:
I have added the validation to all input data I am capturing through those 4 routes I have used. 
and thank you to your valuable feedback, @musllim and @shyakadavis and I hope you are going to review again.